File: 19-07-2024_Count_Smaller_Elements.py
class Solution:
    def constructLowerArray(self, arr):
        n = len(arr)
        ans = [0] * n
        
        for i in range(n):
            count = 0
            for j in range(i + 1, n):
                if arr[j] < arr[i]:
                    count += 1
            ans[i] = count
        
        return ans

File: 19-07-2024_Count_Smaller_Elements.java
class Solution {
    public int[] constructLowerArray(int[] arr) {
        int n = arr.length;
        int[] ans = new int[n];
        
        for (int i = 0; i < n; i++) {
            int count = 0;
            for (int j = i + 1; j < n; j++) {
                if (arr[j] < arr[i]) {
                    count++;
                }
            }
            ans[i] = count;
        }
        
        return ans;
    }
}
